app/src/main/java/com/berniac/vocalwarmup/music/MusicalSymbolParser.java

```diff
@@ -4,8 +4,40 @@
  * Created by Mikhail Lipkovich on 11/17/2017.
  */
 public class MusicalSymbolParser {
-    public static MusicalSymbol parse(String s) {
-        // TODO: Marina extract NoteValue, either Rest or Note
-        return null;
+    public static MusicalSymbol parse(String str) {
+        MusicalSymbol currentSymbol;
+
+        int noteValueEnd = -1;
+        int symbolEnd = str.length();
+        int octaveEnd = str.length();
+
+        for (int i = 0; i < str.length(); i++) {
+            if ((noteValueEnd == -1) & Character.isLetter(str.charAt(i))) {
+                noteValueEnd = i;
+            } else if ((noteValueEnd != -1) & (symbolEnd == str.length())
+                    & !Character.isLetter(str.charAt(i))) {
+                symbolEnd = i;
+            }
+        }
+
+        String noteValue = str.substring(0, noteValueEnd);
+        String symbol = str.substring(noteValueEnd, symbolEnd);
+        String octave = str.substring(symbolEnd, octaveEnd);
+
+        if (octave.equals("")) {
+            octave = "0";
+        }
+
+        if (symbol.equals("N")) {
+            if (symbolEnd != octaveEnd) {
+                throw new IllegalArgumentException("Rest shouldn't contain octave. Octave: " +
```

Comment:
Probably it makes sense to add the whole string instead of octave for all messages. Imagine how would you identify the reason of error if you see such a message in logs